components/more-less/more-less.js

```diff
@@ -0,0 +1,133 @@
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { LitElement, html, css } from 'lit-element/lit-element.js';
+import { LocalizeMixin } from '../localize/localize-mixin.js';
+import '../button/button-subtle.js';
+import 'd2l-icons/d2l-icon.js';
+import 'd2l-icons/tier1-icons.js';
+
+export class D2LMoreLess extends LocalizeMixin(LitElement)  {
+
+	static get properties() {
+		return {
+			expanded: { type: Boolean, reflect: true },
+			icon: { type: String, reflect: true },
+			text: { type: String, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+			}`;
+	}
+
+	constructor() {
+		super();
+		this.langResources = {
+			'ar': {
+				more: 'المزيد',
+				less: 'أقل'
+			},
+			'en': {
+				more: 'more',
+				less: 'less'
+			},
+			'es': {
+				more: 'más',
+				less: 'menos'
+			},
+			'fr': {
+				more: 'plus',
+				less: 'moins'
+			},
+			'ja': {
+				more: 'より多い',
+				less: 'より少ない'
+			},
+			'ko': {
+				more: '더 보기',
+				less: '축소'
+			},
+			'nl': {
+				more: 'meer',
+				less: 'minder'
+			},
+			'pt': {
+				more: 'mais',
+				less: 'menos'
+			},
+			'sv': {
+				more: 'mer',
+				less: 'mindre'
+			},
+			'tr': {
+				more: 'diğer',
+				less: 'daha az'
+			},
+			'zh': {
+				more: '更多',
+				less: '更少'
+			},
+			'zh-tw': {
+				more: '較多',
+				less: '較少'
+			}
+		};
+
+		this.expanded = false;
+	}
+
+	render() {
+		return html`
+			<div class="more-less-content"><slot></slot></div>
+			<div class="more-less-blur"></div>
+			<d2l-button-subtle
+				class="more-less-toggle"
+				icon="${ifDefined(this.icon)}"
```

Comment:
I am wondering if the text and icon change can be handled by using:
```
icon="${ifDefined(this.expanded ? 'd2l-tier1:chevron-up' : 'd2l-tier1:chevron-down')}"
```
Alternatively, `__computeIcon` could be called.  I suspect that when `expanded` changes, `render` will be called again, thereby making `updated` code not necessary.

Reply:
you're correct, just tried this and it seems to work